torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -319,9 +352,30 @@ void CudaCodeGen::Initialize() {
     os() << dtype.ToCppString() << (buffer_arg.isVar() ? " " : "* ")
          << name_manager()->get_unique_name(var);
   }
+  Var rand_seed;
+  Var rand_offset;
+  if (has_random_) {
+    // TODO: switch to kUint64 when it is available.
+    rand_seed = Var("rand_seed", kInt32);
+    rand_offset = Var("rand_offset", kInt32);
+    std::string uint64_str = "unsigned long long";
+    os() << ", " << uint64_str << " " << rand_seed << ", " << uint64_str << " "
+         << rand_offset;
+  }
   os() << ") {";
-
   os() << std::endl;
+
+  if (has_random_) {
+    Var idx{"idx", kInt32};
+    os() << "int " << idx << " = blockIdx.x*blockDim.x + threadIdx.x;"
+         << std::endl;
+    Var rand_func = Var{"rand", kHandle};
+    printer_->SetRandFunc(rand_func);
```

Comment:
If it's possible I'd set rand_func in the printer's constructor so we don't have to twiddle state.